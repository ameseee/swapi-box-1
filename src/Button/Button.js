import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Button = ({ title, changeCategory, stateCategory, category }) => {
  return (
    <div className={
      category===stateCategory
        ? "button-container button-active"
        : "button-container"
    }
    onClick={() => { changeCategory(category); }}>
      <Link to={title}>{title}</Link>
    </div>
  );
};

Button.propTypes = {
  title: PropTypes.string,
  changeCategory: PropTypes.func,
  stateCategory: PropTypes.string
};

export default Button;
