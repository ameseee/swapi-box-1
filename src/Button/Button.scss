.button-container {
  display: inline;
}

.button-active {
  border: 3px solid yellow;
}
