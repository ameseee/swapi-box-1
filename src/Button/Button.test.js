import React from 'react';
import Button from './Button';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('Button', () => {
  Enzyme.configure({ adapter: new Adapter() });
  const mockFn = jest.fn();
  const wrapper = shallow(<Button
    key={Math.random()}
    title='People'
    category='People'
    changeCategory={mockFn}
    stateCategory='People'
  />);

  it('should be active since statecategory === category', () => {
    const active = wrapper.find('.button-active');

    expect(active.length).toEqual(1);
  });

  it('should be inactive when statecategory !== category', () => {
    const altWrapper = shallow(<Button
      key={Math.random()}
      title='People'
      category='People'
      changeCategory={mockFn}
      stateCategory='People'
    />);
    const button = altWrapper.find('.button-container');
    const active = altWrapper.find('.button-active');

    expect(button.length).toEqual(1);
    expect(active.length).toEqual(1);
  });

  it('should have Link', () => {
    const link = wrapper.find('Link');

    expect(link.length).toEqual(1);
  });

  it('should fire function when clicked', () => {
    const div = wrapper.find('div');

    div.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
