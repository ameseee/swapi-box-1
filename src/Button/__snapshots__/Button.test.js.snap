// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    category="People"
    changeCategory={[Function]}
    stateCategory="People"
    title="People"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="People"
>
        People
</Link>,
      "className": "button-container button-active",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "People",
        "replace": false,
        "to": "People",
      },
      "ref": null,
      "rendered": "People",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="People"
>
          People
</Link>,
        "className": "button-container button-active",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "People",
          "replace": false,
          "to": "People",
        },
        "ref": null,
        "rendered": "People",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
